src/shims/unix/fs.rs

```diff
@@ -1710,9 +1709,9 @@ impl FileMetadata {
     fn from_fd_num<'tcx>(
         ecx: &mut MiriInterpCx<'tcx>,
         fd_num: i32,
-    ) -> InterpResult<'tcx, Option<FileMetadata>> {
+    ) -> InterpResult<'tcx, Result<FileMetadata, Scalar>> {
```

Comment:
Please add a doc comment explaining the meaning of this return type. (Same for the next function.)

Reply:
Not necessary, I can use your new error enum instead

Reply:
Ah, even better :)

Reply:
https://github.com/rust-lang/miri/pull/3990 did this part :)